state-sync/src/coordinator.rs

```diff
@@ -83,8 +83,10 @@ pub(crate) struct StateSyncCoordinator<T> {
     request_manager: RequestManager,
     // Optional sync request to be called when the target sync is reached
     sync_request: Option<SyncRequest>,
-    // Ledger infos in the future that have not been committed yet
-    pending_ledger_infos: PendingLedgerInfos,
+    // If we're a full node syncing to the latest state, this holds the highest ledger info
+    // we know about and are currently syncing to. This allows us to incrementally sync to
+    // ledger infos in storage. Higher ledger infos will only be considered once we sync to this.
+    highest_known_ledger_info: Option<LedgerInfoWithSignatures>,
```

Comment:
I thought we can unify this with sync_request since they're both the target we're trying to sync to

Reply:
@zekun000, the plan is to try and unify them, yes, but let's take one step at a time 😄